### ISSUE TITLE
WIP: Word Count on Touch Bar

### ISSUE THREAD

**Issue by rschiang** [MacDownApp/macdown#1031]
After fiddled around the current master build, I realized there's much potential on providing meaningful touch bar features. (Shoutout to #977, we really need to implement customization!) The word count widget, for example, is an extremely useful feature, but showing it on-screen has always been a distraction. 

Here"s a quick-and-dirty [WIP](https://github.com/rschiang/macdown/tree/word-count-touch-bar) that shows the word count being implemented as a popover:

![Screenshot of the touch bar.](https://user-images.githubusercontent.com/1512231/48738835-2465df00-ec8d-11e8-827d-df4a6b719736.png)

![Popover content.](https://user-images.githubusercontent.com/1512231/48739114-e6b58600-ec8d-11e8-93ce-c639352043af.png)

I'll try polish it up so that the on-screen word count widget would hide itself and vice versa.

**Comment by magnuskahr** [comment]
Ehm, the Touch Bar should be an interaction screen, and not somewhere to read information.

**Comment by alexkaessner** [comment]
@magnuskahr is right. Apple states in it's [macOS HIG](https://developer.apple.com/design/human-interface-guidelines/macos/touch-bar/touch-bar-overview/) the following:

> **Use the Touch Bar as an extension of the keyboard and trackpad, not as a display.** Although technically it’s a screen, the Touch Bar functions as an input device, not a secondary display. The user may glance at the Touch Bar to locate or use a control, but their primary focus is the main screen. The Touch Bar shouldn’t display alerts, messages, scrolling content, static content, or anything else that commands the user’s attention or distracts from their work on the main screen.

Even though I kinda like the idea to implement it for users that keep looking a lot on the keyboard like myself. If we want to implement this, it should definitely be an option for a customization of the Touch Bar!

**Comment by robmatyszewski** [comment]
how to install that to my macdown?

**Comment by rschiang** [comment]
@magnuskahr @alexkaessner You're right, the current implementation is stepping on the borderline as it merely reflects the word count dropdown. It might even be better if it were a plugin; however, there seems to exist no mechanism for plugins to interfere with UI, so it has to be done in the main program as of now.

@robmatyszewski You could pull the branch and build a custom version out of it. It's not an official feature yet.
